android/java/org/chromium/chrome/browser/app/domain/WalletModel.java

```diff
@@ -87,11 +94,39 @@ public void resetServices(Context context, KeyringService keyringService,
                 mAssetRatioService);
         mDappsModel.resetServices(
                 mJsonRpcService, mBraveWalletService, mCryptoModel.getPendingTxHelper());
-        mKeyringModel.resetService(mContext, mKeyringService, braveWalletService);
+        mKeyringModel.resetService(mContext, mKeyringService, braveWalletService, jsonRpcService);
         mMarketModel.resetService(mAssetRatioService);
         init();
     }
 
+    public void setOriginInfo(OriginInfo originInfo) {
+        if (mOriginInfo != null && originInfo != null
+                && mOriginInfo.originSpec.equals(originInfo.originSpec)) {
```

Comment:
I don't know if we may and up in this edge case, but:
⚠️ If `mOriginInfo.originSpec == null` we'd have a crash here.

Reply:
I noticed that though `originSpec` won't be null as it will always come from a dapps site after the completion of loading.